### PR TITLE
Disable workspace AppInsight's private endpoint

### PR DESCRIPTION
**Pull request by tamirkamara**
## What is being addressed

Due to many failures when creating the workspace's private endpoint for app insights, we need to disable it for now while our support case / PG conversation is still going.

In addition this PR disables the Terraform checkpoint mechanism as that doesn't work in an airgap environment. The error is comes to solve is Terraform unable to query provider/plugin versions.

### CODE REVIEW

File: templates/workspaces/base/terraform/azure-monitor/azure-monitor.tf [hunk 1]
```diff
@@ -79,6 +79,7 @@ resource "azurerm_monitor_private_link_scoped_service" "ampls_app_insights" {
 }
 
 resource "azurerm_private_endpoint" "azure_monitor_private_endpoint" {
+  count               = 0 # Remove with https://github.com/microsoft/AzureTRE/issues/2357
```
Comment (yuvalyaron) [inline templates/workspaces/base/terraform/azure-monitor/azure-monitor.tf:82]: how is this related to the issue?
should we mention anything about this block there?

**Review by yuvalyaron — APPROVED** [review]

File: templates/workspaces/base/terraform/azure-monitor/azure-monitor.tf (hunk 1 above)
Comment (tamirkamara) [reply templates/workspaces/base/terraform/azure-monitor/azure-monitor.tf:82]: there 2 things addressed here: the problem with the PE which is happening for some time, and a newer one about the provider (checkpoint) which I also tried to address here since I couldn't get PRs to pass E2Es.
